packages/flutter/lib/src/material/search.dart

```diff
@@ -519,33 +535,32 @@ class _SearchPageState<T> extends State<_SearchPage<T>> {
       scopesRoute: true,
       namesRoute: true,
       label: routeName,
-      child: Scaffold(
-        appBar: AppBar(
-          backgroundColor: theme.primaryColor,
-          iconTheme: theme.primaryIconTheme,
-          textTheme: theme.primaryTextTheme,
-          brightness: theme.primaryColorBrightness,
-          leading: widget.delegate.buildLeading(context),
-          title: TextField(
-            controller: widget.delegate._queryTextController,
-            focusNode: focusNode,
-            style: theme.textTheme.headline6,
-            textInputAction: widget.delegate.textInputAction,
-            keyboardType: widget.delegate.keyboardType,
-            onSubmitted: (String _) {
-              widget.delegate.showResults(context);
-            },
-            decoration: InputDecoration(
-              border: InputBorder.none,
-              hintText: searchFieldLabel,
-              hintStyle: searchFieldStyle,
+      child: Theme(
+        data: theme,
+        child: Scaffold(
+          appBar: AppBar(
+            backgroundColor: theme.primaryColor,
+            iconTheme: theme.primaryIconTheme,
+            textTheme: theme.primaryTextTheme,
+            brightness: theme.primaryColorBrightness,
```

Comment:
Ok, really sorry about this, but I was wrong here. We do need these removed, or apps won't be able to override them with `AppBarTheme` settings (either globally or in the returned theme from the delegates method). This will break the sample code in #45498.

So please remove this again. We can fix the tests below.